app/src/main/java/com/g11x/checklistapp/NavigationActivity.java

```diff
@@ -73,12 +75,19 @@ protected void onCreate(Bundle savedInstanceState) {
     final Activity thisActivity = NavigationActivity.this;
     baseLanguageChangeListener = new AppPreferences.LanguageChangeListener(NavigationActivity.this) {
       @Override
-      public void onChanged(String newValue) {
-        thisActivity.recreate();
+      void onChanged(@NonNull Language newLangauge) {
+        Intent intent = getIntent();
+        finish();
+        startActivity(intent);
```

Comment:
`finish()` eventually leads to `onDestroy()`. I'm starting a new instance of the same activity and relying on the normal activity transitions to provide a nicer UX than before.  Instead, I could have notified the adapter of the changes, but since the selection of a new language is effectively an Android Config change I wanted the whole activity (incl NavBar) to re-pull resources.